package.json

```diff
@@ -101,5 +104,14 @@
     "type-fest": "^4.10.3",
     "typescript": "~5.2.2"
   },
+  "lint-staged": {
```

Comment:
I have never used `lint-staged` (`nx affected` does similar job, probably even better), however, why is it required to duplicate the config for TS and MD/YAML? ESLint skips files that are not matched by the rules. 🤔

Reply:
I don’t want to force you to follow my preferences, however, IMHO in CI, we should only _check_ the style (ESLint and Prettier), never fix it. I get it that it might be easier for devs, however, I still think that all the changes (except for release-related changes) should be done by the dev, incl the code styling and formatting. If there are any issues (code, test, lint or format), it should be fixed by the dev.

Reply:
this isn't CI, this is pre-push

Reply:
Okay, if it is pre-push, linting and formatting should be already checked in the CI before we can merge the PR. :thinking:

Reply:
yeah, this ensures that won't be a problem when the commit gets up to the PR. 

husky can't protect web-based commits anyway, so the CI action to check the above are still needed to protect anyone
- making web-based commits
- bypassing pre-commit checks (`git commit -n`)